gluten-core/src/main/scala/org/apache/spark/util/GlutenShutdownManager.scala

```diff
@@ -16,14 +16,28 @@
  */
 package org.apache.spark.util
 
+import org.apache.gluten.backendsapi.BackendsApiManager
+
+import org.apache.commons.lang3.{JavaVersion, SystemUtils}
+
 object GlutenShutdownManager {
 
   def addHook(hook: () => Unit): AnyRef = {
     ShutdownHookManager.addShutdownHook(ShutdownHookManager.DEFAULT_SHUTDOWN_PRIORITY)(hook)
   }
 
   def addHookForLibUnloading(hook: () => Unit): AnyRef = {
-    ShutdownHookManager.addShutdownHook(ShutdownHookManager.SPARK_CONTEXT_SHUTDOWN_PRIORITY)(hook)
+    ShutdownHookManager.addShutdownHook(ShutdownHookManager.SPARK_CONTEXT_SHUTDOWN_PRIORITY) {
+      if (
+        BackendsApiManager.getBackendName == "velox" && SystemUtils.isJavaVersionAtLeast(
```

Comment:
`if (BackendsApiManager.getBackendName == "...")`

Seems like a writing we should avoid in core module's code. Let's figure out if there is a better way.